ports/webassembly/objpyproxy.js

```diff
@@ -169,30 +169,33 @@ const py_proxy_handler = {
             return null;
         }
 
-        if (prop === Symbol.iterator) {
-            // Get the Python object iterator, and return a JavaScript generator.
-            const iter_ref = Module.ccall(
-                "proxy_c_to_js_get_iter",
-                "number",
-                ["number"],
-                [target._ref],
-            );
-            return function* () {
-                const value = Module._malloc(3 * 4);
-                while (true) {
-                    const valid = Module.ccall(
-                        "proxy_c_to_js_iternext",
-                        "number",
-                        ["number", "pointer"],
-                        [iter_ref, value],
-                    );
-                    if (!valid) {
-                        break;
+        if (typeof prop !== "string") {
```

Comment:
Is it worth moving this if-block up to the top of this `get()` function?  Then if it fails we know `prop` is a string and it makes sense to test it against known strings like `_ref` and `then`.

This could maybe make the JS code slightly faster?

Reply:
yes and no ... direct strict equality comparison is faster than `typeof` so I was trying to fallback gracefully for those `_ref` and `then` use cases ... true that `then` is an indirect edge-case but I wasn't sure about how much `_ref` access is needed so that I think check against `_ref` should be on top, for perf reasons, then we can check for `typeof` but at that point it'd be just moving `then` check after ... not much gain?